src/Bicep.Core/Navigation/IArtifactReferenceSyntax.cs

```diff
@@ -16,4 +18,6 @@ public interface IArtifactReferenceSyntax
     SyntaxBase? Path { get; }
 
     ArtifactType GetArtifactType();
+
+    ResultWithDiagnostic<string> ResolveArtifactPath(RootConfiguration config);
```

Comment:
It seems odd to pass a template's configuration to a syntax node. `RootConfiguration` is a bit too nebulous to be passing around outside of service constructors -- per [Hyrum's law](https://www.hyrumslaw.com/), this is an invitation for syntax nodes to do something wacky with the extra data being passed in -- so at the very least this would need to be narrowed down to only the relevant part of the configuration. Another odd thing is that it is not consistently used; module declarations (which use aliases declared in the configuration) just ignore the configuration object, whereas provider declarations resolve aliases inline.

Would it make sense to move this logic somewhere else (maybe within `ModuleDispatcher` where we're already handling aliases)?

Reply:
> I couldn't move this logic to the ModuleDispatcher because the resolution of the artifact details is necessary in earlier processing stages

I'm not sure that's true. The only place I saw this method being called was [within `ModuleDispatcher`](https://github.com/Azure/bicep/blob/23ebf70911965b15b08a759d2734d93117935cf9/src/Bicep.Core/Registry/ModuleDispatcher.cs#L86-L92); was there somewhere else it is being used?

> but for now, given that this continues to be an implementation that is subject to change and considering that it's important to make progress on this feature I would like to address this concern as a separate topic to be handled in a different change request.

If this is technical debt that you expect to be impermanent, would it make sense to reduce the number of places you will need to change when the debt is paid down? Instead of adding this method to the interface, adding it directly to `ProviderDeclarationSyntax` and then adding an `if (syntax is ProviderDeclarationSyntax pds)` guard where you need the custom logic would keep it small and targeted.

Reply:
I agree and would personally vote to move this out of the syntax class. We haven't been very strict about this, but I think it's worth keeping syntax logic minimal.

Unless I'm missing something, there's nothing to stop this being a static helper method.

Reply:
Ok, I will remove this function from the interface and handle it in `ModuleDispatcher`